### DISCUSSION TITLE
Disable Fields But Still Return The Data?

### DISCUSSION THREAD

**Discussion by fadi-george** [react-hook-form/react-hook-form#2686] [Q&A] [answered]
Hello,

I saw in the recent updates that some things were changed. Before I was relying on data being passed back in the handleSubmit even when the field was disabled. Now this is no longer the case.

What would be the best approach for handling disabled fields? Should I not register disabled fields and just spread my own data in the handleSubmit function?

Sandbox:
https://codesandbox.io/s/react-hook-form-v6-validationschema-forked-zi9eh?file=/src/index.js

**Comment by bluebill1049** [comment]
can you use `getValues()`?

**Comment by fadi-george** [comment, marked as answer]
I suppose for now I'll have to remove that yup required check and re-work my logic to not need all fields passed.
If I call getValues, it will return undefined for disabled fields.

**Reply by bluebill1049** [reply]
yes, the `disabled` field will not be part of the input submit, you can change to read-only tho. that's the difference.

**Reply by fadi-george** [reply]
Ah `readOnly`, didn't think of that 🤔. That should work.
Thanks!

**Reply by bluebill1049** [reply]
no worries, just trying to follow html standard, disabled is something that should be excluded in the final submit.

**Reply by fadi-george** [reply]
It was my fault for using things incorrectly. Thanks for your quick help. It's an awesome library!
